docs/smartcontracts.md

```diff
@@ -0,0 +1,172 @@
+
+## Taquito's Smart Contract Abstraction
+
+Taquito assists developers by reading the Michelson code for a given contract from the blockchain. Based on the retrieved Michelson code, Taquito generates a `contract` javascript object with methods and storage that correspond to the contacts Michelson entrypoints, storage definitions and values.
+
+## The Counter Contract
+
+In this guide, we use a straightforward "counter" smart contract to illustrate how Taquito works.
+
+The counter contract has two entrypoints named `increment` and `decrement`.  These entrypoints are used by Taquito to generate corresponding javascript methods that are available to the developer.
+
+The counter contracts storage is a simple integer that gets increased or decreased based on the calls to the entrypoints.
+
+### Counter Contract in CameLIGO
+
+```ocaml
+type storage = int
+
+(* variant defining pseudo multi-entrypoint actions *)
+
+type action =
+| Increment of int
+| Decrement of int
+
+let add (a,b: int * int) : int = a + b
+let sub (a,b: int * int) : int = a - b
+
+(* real entrypoint that re-routes the flow based on the action provided *)
+
+let main (p,s: action * storage) =
+ let storage =
+   match p with
+   | Increment n -> add (s, n)
+   | Decrement n -> sub (s, n)
+ in ([] : operation list), storage
+
+```
+
+[View the counter contract in the Ligo WebIDE][2]
+
+### Counter Contract Michelson source code
+
+```
+{ parameter (or (int %decrement) (int %increment)) ;
+  storage int ;
+  code { DUP ;
+         CDR ;
+         DIP { DUP } ;
+         SWAP ;
+         CAR ;
+         IF_LEFT
+           { DIP { DUP } ;
+             SWAP ;
+             DIP { DUP } ;
+             PAIR ;
+             DUP ;
+             CAR ;
+             DIP { DUP ; CDR } ;
+             SUB ;
+             DIP { DROP 2 } }
+           { DIP { DUP } ;
+             SWAP ;
+             DIP { DUP } ;
+             PAIR ;
+             DUP ;
+             CAR ;
+             DIP { DUP ; CDR } ;
+             ADD ;
+             DIP { DROP 2 } } ;
+         NIL operation ;
+         PAIR ;
+         DIP { DROP 2 } } }
+```
+
+## Loading the contract in Taquito
+
+To load the contract from the Tezos Blockchain, we use the `Tezos.contract.at` method.
+We can inspect the contract methods and data types using the `c.parameterSchema.ExtractSignatures()` method.
+
+The following example shows how to load the contact, and view the methods on that contract.
+
+```js live noInline
+Tezos.setProvider({ rpc: 'https://api.tez.ie/rpc/babylonnet' });
+
+Tezos.contract.at('KT1LjpCPTqGajeaXfLM3WV7csatSgyZcTDQ8')
+.then(c => {
+ 
+    let methods = c.parameterSchema.ExtractSignatures()
+
+    render(JSON.stringify(methods))
+    
+    //console.log(c.methods.increment(2).toTransferParams())
```

Comment:
Is it intended to leave these as commented lines?

Reply:
derp :laughing: